tests-e2e/argocd/argocd_instance_test.go

```diff
@@ -126,5 +126,93 @@ var _ = Describe("Standalone ArgoCD instance E2E tests", func() {
 				))
 
 		})
+
+		It("should update existing ArgoCD resources, create an application, wait for it to be installed and synced", func() {
+
+			By("creating ArgoCD resource")
+			ctx := context.Background()
+			log := log.FromContext(ctx)
+
+			config, err := fixture.GetSystemKubeConfig()
+			Expect(err).ToNot(HaveOccurred())
+			apiHost := config.Host
+
+			k8sClient, err := fixture.GetKubeClient(config)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = argocdv1.ReconcileNamespaceScopedArgoCD(ctx, argocdCRName, argocdNamespace, k8sClient, log)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("ensuring ArgoCD service resource exists")
+			argocdInstance := &apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: argocdCRName + "-server", Namespace: argocdNamespace},
+			}
+
+			Eventually(argocdInstance, "60s", "5s").Should(k8s.ExistByName(k8sClient))
+			Expect(err).ToNot(HaveOccurred())
+
+			By("ensuring ArgoCD resource exists in kube-system namespace")
+			err = argocdv1.SetupArgoCD(ctx, apiHost, argocdNamespace, k8sClient, log)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("ensuring the SetupArgoCD function works even if resources already exist in kube-system namespace")
```

Comment:
Is it possible to refactor and maybe common up this test case with at least the previous one?  From what I see this block of code is the only difference.

Reply:
Fixed